src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -378,7 +393,8 @@ private void AddSemanticRange(SyntaxNode node, int semanticKind)
         {
             if (node is null)
             {
-                throw new ArgumentNullException(nameof(node));
+                // There are some race conditions under which the nodes might be null as someone types `="literal stuff` being one observed example.
+                return;
```

Comment:
https://github.com/dotnet/aspnetcore-tooling/pull/2981. I think this was recently exposed by a change in the parse tree? I'm actually going to remove this from this PR because I've got it fixed better in the other one.